bundles/binding/org.openhab.binding.gc100ir/src/main/java/org/openhab/binding/gc100ir/lib/GC100IRConnection.java

```diff
@@ -0,0 +1,193 @@
+     * deviceType to local variables.
+     *
+     * @param inetAddress
+     *            an Object of InetAddress
+     * @param ipAddressString
+     *            an String of InetAddress
+     * @param port
+     *            an integer value of port
+     * @param module
+     *            an integer value of module
+     * @param connector
+     *            an integer value of connector
+     * @param deviceType
+     *            a String value of deviceType
+     * @throws IOException
+     */
+    GC100IRConnection(InetAddress ipAddress, String ipAddressString, int port, int module, int connector,
+            String deviceType) throws IOException {
+        this.inetAddress = ipAddress;
+        this.ipAddressString = ipAddressString;
+        this.port = port;
+        this.module = module;
+        this.connector = connector;
+        this.deviceType = deviceType;
+    }
+
+    /**
+     * Constructor. Initializes the inetAddress, port, module, connector and
+     * deviceType to local variables.
+     *
+     * @param ipAddress
+     *            an Object of InetAddress
+     * @param ipAddressString
+     *            String representation of the ip address
+     * @param port
+     *            an integer value of port
+     * @param module
+     *            an integer value of module
+     * @param connector
+     *            an integer value of connector
+     * @param deviceType
+     *            a String value of deviceType
+     * @param socket
+     *            an Object of Socket
+     * @throws IOException
+     */
+    GC100IRConnection(InetAddress ipAddress, String ipAddressString, int port, int module, int connector,
+            String deviceType, Socket socket) throws IOException {
+        this.inetAddress = ipAddress;
+        this.ipAddressString = ipAddressString;
+        this.port = port;
+        this.module = module;
+        this.connector = connector;
+        this.deviceType = deviceType;
+        this.socket = socket;
+        this.isConnected = true;
+    }
+
+    /**
+     * Checks if GC100IRConnection is connected.
+     *
+     * @return true if connected false otherwise.
+     */
+    boolean isConnected() {
+        return isConnected;
+    }
+
+    /**
+     * Returns IP address in string format.
+     *
+     * @return an IP address string
+     */
+    String getIPAddressString() {
+        return ipAddressString;
+    }
+
+    /**
+     * Connect to the TCP/IP socket.
+     *
+     * @return a boolean whether connection is successful or not.
+     * @throws IOException
+     */
+    boolean connect() throws IOException {
+        if (isConnected) {
+            return true;
+        }
+
+        try {
+            socket = new Socket(inetAddress, port);
+
+            if (socket.isConnected()) {
+                isConnected = true;
+                return true;
+            }
+        } catch (Exception e) {
+            logger.error("Bind Exception Already Connected ", e);
```

Comment:
Catch the specific exception(s) and log them consistently with what they are.  Also, as is, this method does not throw IOException.